### PR TITLE
@W-15316975: [Android] Update native login sample app to include User Registration flow

### PR DESCRIPTION
**Pull request by JohnsonEricAtSalesforce**
🎸 _Review For Review!_ 🥁

Now we have headless registration via the Salesforce Identity API as well.  This will look very familiar to reviewers of other recent headless identity flow pull requests.

### CODE REVIEW

File: AndroidNativeLoginTemplate/app/src/main/java/com/salesforce/androidnativelogintemplate/MainApplication.kt [hunk 1]
```diff
@@ -142,14 +142,17 @@ class MainApplication : Application() {
...
-         * Executes the Google reCAPTCHA client for a new login action token.
+         * Executes the Google reCAPTCHA client for a new token for the
+         * specified action.
+         * @param reCaptchaAction A Google reCAPTCHA SDK reCAPTCHA action
          * @param completion The function to invoke with the new token or null
          * if the token could not be obtained
          */
-        internal fun executeLoginAction(
+        internal fun executeReCaptchaAction(
+            reCaptchaAction: RecaptchaAction,
```
Comment (JohnsonEricAtSalesforce) [inline AndroidNativeLoginTemplate/app/src/main/java/com/salesforce/androidnativelogintemplate/MainApplication.kt:152]: Now the sample app demonstrates how to use appropriate reCAPTCHA actions as documented by Google.

File: AndroidNativeLoginTemplate/app/src/main/java/com/salesforce/androidnativelogintemplate/NativeLogin.kt [hunk 2]
```diff
@@ -157,33 +163,13 @@ class NativeLogin : ComponentActivity() {
...
-                                    return@LoginView false
-                                }
-
-                                Success -> {
-                                    finish()
-                                }
-                            }
-
-                            return@LoginView true
+                            return@LoginView nativeLoginManager.login(
```
Comment (JohnsonEricAtSalesforce) [inline AndroidNativeLoginTemplate/app/src/main/java/com/salesforce/androidnativelogintemplate/NativeLogin.kt:166]: This was becoming redundant, so I made a quick utility out of it.

File: AndroidNativeLoginTemplate/app/src/main/java/com/salesforce/androidnativelogintemplate/NativeLogin.kt [hunk 3]
```diff
@@ -512,15 +492,21 @@ class NativeLogin : ComponentActivity() {
                         elevation = CardDefaults.cardElevation(defaultElevation = 6.dp)
                     ) {
                         Spacer(modifier = Modifier.height(25.dp))
-                        Image(
-                            painter = painterResource(id = sf__salesforce_logo),
-                            colorFilter = ColorFilter.tint(colorScheme.primary),
-                            contentDescription = "",
-                            modifier = Modifier.align(CenterHorizontally),
-                        )
+                        if (identityFlowLayoutTypeActive != StartRegistration) {
```
Comment (JohnsonEricAtSalesforce) [inline AndroidNativeLoginTemplate/app/src/main/java/com/salesforce/androidnativelogintemplate/NativeLogin.kt:495]: There's an issue with the IME layout of the native login view where tall layouts don't correctly scroll to the focused element.  I've been trying to fix it, but it certainly looks worth its own bug in the interest of cycle time and small work units.  In the meantime, it's not so bad if the rather large SF logo isn't displayed for start registration.

I'll get that new work item in for a better IME layout.

File: AndroidNativeLoginTemplate/app/src/main/java/com/salesforce/androidnativelogintemplate/NativeLogin.kt [hunk 4]
```diff
@@ -576,8 +570,163 @@ class NativeLogin : ComponentActivity() {
     }
 
     @Composable
-    fun StartPasswordResetInput() {
+    fun StartRegistrationInput() {
```
Comment (JohnsonEricAtSalesforce) [inline AndroidNativeLoginTemplate/app/src/main/java/com/salesforce/androidnativelogintemplate/NativeLogin.kt:573]: The new layouts are very similar in style to the existing ones.

**Review by JohnsonEricAtSalesforce — COMMENTED** [review]
The new code for registration is very similar to existing patterns from login, password-less login and forgot password, so there's just a few quick walkthrough comments here 😎

File: AndroidNativeLoginTemplate/app/src/main/java/com/salesforce/androidnativelogintemplate/NativeLogin.kt (hunk 3 above)
Comment (JohnsonEricAtSalesforce) [reply AndroidNativeLoginTemplate/app/src/main/java/com/salesforce/androidnativelogintemplate/NativeLogin.kt:495]: The follow-up is `W-15811064`.

**Review by brandonpage — APPROVED** [review]
👍 Would be nice to get some screenshots whenever PRs are heavy on the UI code though.

**Review by wmathurin — APPROVED** [review]

**Pull request merged**